flytekit/core/interface.py

```diff
@@ -267,27 +267,34 @@ def _change_unrecognized_type_to_pickle(t: Type[T]) -> Type[T]:
     return t
 
 
-def transform_signature_to_interface(signature: inspect.Signature, docstring: Optional[Docstring] = None) -> Interface:
+def transform_function_to_interface(fn: Callable, docstring: Optional[Docstring] = None) -> Interface:
     """
     From the annotations on a task function that the user should have provided, and the output names they want to use
     for each output parameter, construct the TypedInterface object
 
     For now the fancy object, maybe in the future a dumb object.
+
     """
-    outputs = extract_return_annotation(signature.return_annotation)
-    for k, v in outputs.items():
-        outputs[k] = _change_unrecognized_type_to_pickle(v)
-    inputs = OrderedDict()
-    for k, v in signature.parameters.items():
-        annotation = v.annotation
-        default = v.default if v.default is not inspect.Parameter.empty else None
-        # Inputs with default values are currently ignored, we may want to look into that in the future
-        inputs[k] = (_change_unrecognized_type_to_pickle(annotation), default)
+    type_hints = typing.get_type_hints(fn)
+    signature = inspect.signature(fn)
+    defaults = {
+        arg: param.default if param.default is not inspect.Parameter.empty else None
+        for arg, param in signature.parameters.items()
+    }
+
+    # Inputs with default values are currently ignored, we may want to look into that in the future
+    inputs[k] = (_change_unrecognized_type_to_pickle(annotation), default)
+
+    outputs = extract_return_annotation(type_hints.get("return", None))
+    inputs = {
+        arg: (_change_unrecognized_type_to_pickle(type_hints.get(arg, inspect.Parameter.empty)), default)
+        for arg, default in defaults.items()
+    }
```

Comment:
can we keep the original two loops after we set `type_hints` and `signature` instead?

Reply:
Ah yeah I had removed some important behavior on outputs (sorry, merged this in from a change I was doing on 0.22.2); I updated, but if you're also suggesting using a for loop instead of a dict comprehension happy to make that change specifically also :)

Reply:
nah, it's pretty clear what's happening after the last update!

Reply:
Okay, made the change a bit less intrusive and kept as much of the previous loops as possible :) resolving this.